p2p/exceptions.py

```diff
@@ -1,54 +1,95 @@
-class AuthenticationError(Exception):
+class BaseP2PError(Exception):
+    """
+    Base error class for all p2p errors.
+    """
     pass
 
-
-class DecryptionError(Exception):
+class AuthenticationError(BaseP2PError):
```

Comment:
Judging from the callsites it seemed to me as if this and `DecryptionError` could get merged into one. Thoughts?

Reply:
Seems like it

Reply:
Alright, will update the PR.